docs/es/week12/12-3.md

```diff
@@ -0,0 +1,546 @@
+
+
+<!-- ### Self Attention (I) -->
+### Auto Atención (I)
+
+<!-- Consider a set of $t$ input $\boldsymbol{x}$'s: -->
+Consideremos un conjunto de $t$ entradas $\boldsymbol{x}$'s:
+
+$$
+\lbrace\boldsymbol{x}_i\rbrace_{i=1}^t = \lbrace\boldsymbol{x}_1,\cdots,\boldsymbol{x}_t\rbrace
+$$
+
+<!-- where each $\boldsymbol{x}_i$ is an $n$-dimensional vector. Since the set has $t$ elements, each of which belongs to $\mathbb{R}^n$, we can represent the set as a matrix $\boldsymbol{X}\in\mathbb{R}^{n \times t}$. -->
+donde cada $\boldsymbol{x}_i$ es un vector $n$-dimensional. Puesto que el conjunto tiene $t$ elementos, y cada uno de ellos pertenece a $\mathbb{R}^n$, podemos representar el conjunto como una matriz $\boldsymbol{X}\in\mathbb{R}^{n \times t}$.
+
+<!-- With self-attention, the hidden representation $h$ is a linear combination of the inputs: -->
+En el caso de la auto atención, la representación oculta $h$ es una combinación lineal de las entradas:
+
+$$
+\boldsymbol{h} = \alpha_1 \boldsymbol{x}_1 + \alpha_2 \boldsymbol{x}_2 + \cdots +  \alpha_t \boldsymbol{x}_t
+$$
+
+<!-- Using the matrix representation described above, we can write the hidden layer as the matrix product: -->
+Usando la representación matricial descrita arriba, podemos escribir la capa oculta como el producto matricial:
+
+$$
+\boldsymbol{h} = \boldsymbol{X} \boldsymbol{a}
+$$
+
+<!-- where $\boldsymbol{a} \in \mathbb{R}^n$ is a column vector with components $\alpha_i$. -->
+donde $\boldsymbol{a} \in \mathbb{R}^n$ es un vector columna con componentes $\alpha_i$.
+
+<!-- Note that this differs from the hidden representation we have seen so far, where the inputs are multiplied by a matrix of weights. -->
+Nótese que esto difiere de la representación oculta que hemos visto hasta ahora, donde las entradas son multiplicadas por una matriz de pesos.
+
+<!-- Depending on the constraints we impose on the vector $\vect{a}$, we can achieve hard or soft attention. -->
+Dependiendo de las restricciones que impongamos en el vector $\vect{a}$, podemos conseguir atención intensa o atención ténue.
+
+
+<!-- #### Hard Attention -->
+#### Atención Intensa
+
+<!-- With hard-attention, we impose the following constraint on the alphas: $\Vert\vect{a}\Vert_0 = 1$. This means $\vect{a}$ is a one-hot vector. Therefore, all but one of the coefficients in the linear combination of the inputs equals zero, and the hidden representation reduces to the input $\boldsymbol{x}_i$ corresponding to the element $\alpha_i=1$. -->
+En la atención intensa, imponemos las siguientes restricciones en las alfas: $\Vert\vect{a}\Vert_0 = 1$. Esto significa que $\vect{a}$ es un vector con codificación "one-hot". Por lo tanto, todos los coeficientes (con excepción de uno) son iguales a cero en la combinación lineal de las entradas, y la representación interna se reduce a la entrada $\boldsymbol{x}_i$ que corresponde al elemento $\alpha_i=1$.
```

Comment:
Is there a better translation for "one-hot encoding" in Spanish?

Reply:
"uno-caliente"?

Reply:
I'd personally leave it as "one-hot". When I was taking CS back at college I remember having seen it that way.

Reply:
I agree with @xcastilla, the term "uno-caliente" sounds very strange in Spanish.

Reply:
Fine with me. I'm not a native speaker.